auv_navigation/auv_mapping/scripts/gate_trajectory_publisher.py

```diff
@@ -17,134 +17,224 @@ def __init__(self):
-            distance_closer = math.sqrt(new_x_closer**2 + new_y_closer**2)
-
-            # Determine which frame is closer to odom
-            if distance_closer < distance_away:
-                entrance_x, entrance_y, entrance_z = (
-                    new_x_closer,
-                    new_y_closer,
-                    selected_z,
-                )
-                entrance_quat = quat_closer
-                exit_x, exit_y, exit_z = new_x_away, new_y_away, selected_z
-                exit_quat = quat_away
-            else:
-                entrance_x, entrance_y, entrance_z = new_x_away, new_y_away, selected_z
-                entrance_quat = quat_away
-                exit_x, exit_y, exit_z = new_x_closer, new_y_closer, selected_z
-                exit_quat = quat_closer
-
-            # Create TransformStamped messages for entrance and exit
-            t_entrance = geometry_msgs.msg.TransformStamped()
-            t_entrance.header.stamp = rospy.Time.now()
-            t_entrance.header.frame_id = self.world_frame
-            t_entrance.child_frame_id = self.new_frame_closer
-            t_entrance.transform.translation.x = entrance_x
-            t_entrance.transform.translation.y = entrance_y
-            t_entrance.transform.translation.z = entrance_z
-            t_entrance.transform.rotation.x = entrance_quat[0]
-            t_entrance.transform.rotation.y = entrance_quat[1]
-            t_entrance.transform.rotation.z = entrance_quat[2]
-            t_entrance.transform.rotation.w = entrance_quat[3]
-
-            t_exit = geometry_msgs.msg.TransformStamped()
-            t_exit.header.stamp = rospy.Time.now()
-            t_exit.header.frame_id = self.world_frame
-            t_exit.child_frame_id = self.new_frame_away
-            t_exit.transform.translation.x = exit_x
-            t_exit.transform.translation.y = exit_y
-            t_exit.transform.translation.z = exit_z
-            t_exit.transform.rotation.x = entrance_quat[0]
-            t_exit.transform.rotation.y = entrance_quat[1]
-            t_exit.transform.rotation.z = entrance_quat[2]
-            t_exit.transform.rotation.w = entrance_quat[3]
-
-            # Send transforms using the service
-            self.send_transform(t_entrance)
-            self.send_transform(t_exit)
+            exit_position, exit_quat = (shifted_position_outward, shifted_quat_outward)
+
+        # Finally, create the entrance and exit poses
+        entrance_pose = Pose()
+        entrance_pose.position = Point(*entrance_position)
+        entrance_pose.orientation = Quaternion(*entrance_quat)
 
+        exit_pose = Pose()
+        exit_pose.position = Point(*exit_position)
+        exit_pose.orientation = Quaternion(*exit_quat)
+
+        return entrance_pose, exit_pose
+
+    def create_trajectory_frames(self) -> None:
+        """
+        Look up the current transforms, compute entrance and exit transforms, and send them.
```

Comment:
Look up the current transforms, compute entrance and exit transforms, and broadcast them.